### PR TITLE
fix: Fix Genesis State for Data Certificate and Add Sentinel Error

### PR DESCRIPTION
**Pull request by inchori**
Fix ```InitGenesis``` and ```ExportGenesis``` function in ```genesis.go``` for Data Certificate.

Add sentinel error in ```types/errors.go``` and apply sentinel errors to ```deal.go``` and ```deal_test.go```.

### CODE REVIEW

**Review by gyuguen — APPROVED** [review]
LGTM 👍

**Comment by gyuguen** [comment]
Althought it is not relate to this issue (As @hansol-medi said), there is no place status where the Deal is `COMPLETED`.
I think it would be changed at the time of `SellData`

File: x/market/genesis.go [hunk 1]
```diff
@@ -23,12 +26,14 @@ func ExportGenesis(ctx sdk.Context, k keeper.Keeper) *types.GenesisState {
 
 	dealsMap := make(map[uint64]*types.Deal)
 
-	//TODO: Implement GetDealsList and Mapping to dealsMap
+	dataCertificateMap := make(map[string]*types.DataValidationCertificate)
+	//TODO: Implement GetDealsList and GetDataCertificateByDealId Mapping to dealsMap and dataCertificateMap.
 	//for _, deal := k.GetDeals(ctx) {
 	//}
```
[the remark this answers is not in the document]
Comment (inchori) [reply x/market/genesis.go:32]: Yes, I'm agree with your review. I'll work in next PR.

File: proto/panacea/market/v2/genesis.proto [hunk 2]
```diff
@@ -5,9 +5,11 @@ option go_package = "github.com/medibloc/panacea-core/x/market/types";
...
 import "panacea/market/v2/deal.proto";
 import "google/protobuf/any.proto";
+import "panacea/market/v2/tx.proto";
 
 // GenesisState defines the market module's genesis state.
 message GenesisState {
   map<uint64, Deal> deals = 1;
-  uint64 next_deal_number = 2;
+  map<string, DataValidationCertificate> data_certificate = 2;
+  uint64 next_deal_number = 3;
```
[the remark this answers is not in the document]
Comment (youngjoon-lee) [reply proto/panacea/market/v2/genesis.proto:14]: I agree with gyuguen. Also, it's okay to put the `data_certificate` after the `next_deal_number` (`data_certificate = 3`).

Plus, please use a plural word: `data_certificates`.

**Review by youngjoon-lee — APPROVED** [review]
lgtm

**Pull request merged**